src/plugins/intel_gna/backend/make_pwl.cpp

```diff
@@ -363,43 +334,100 @@ void make_gna_pwl(const DnnActivation&  fun,
             print_segment((int32_t)(gna_pwl[i + 1].xBase & XBASEMASK) / in_scale, gna_pwl[i + 1].yBase / out_scale, 1.0);
             break;
         }
-        case kActPow: {
-            float pow_exponent = fun.args.pow.exponent;
-            IE_ASSERT(pow_exponent != 1.0f);
-            if (pow_exponent == 0.0f) {
-                int32_t x_lower = INT32_MIN;
-                int32_t x_upper = INT32_MAX;
-                int16_t y_lower = FLOAT_TO_INT16(1 * out_scale);
-                int16_t y_upper = y_lower;
-                auto n_segments = 2;
-
-                gna_pwl.resize(n_segments);
-
-                gna_pwl[0].xBase = INT32_MIN & XBASEMASK;  // zero out the 2 lsb
-                gna_pwl[0].yBase = y_lower;
-                gna_pwl[0].slope = 0;
-                print_segment(gna_pwl[0].xBase / in_scale, gna_pwl[0].yBase / out_scale, 0.0);
-
-                gna_pwl[1].xBase = x_lower & XBASEMASK;  // zero out the 2 lsb
-                gna_pwl[1].yBase = y_lower;
-                double slope = (static_cast<double>(y_upper - y_lower) / out_scale) / (static_cast<double>(x_upper - x_lower) / in_scale);
-                s = gna_slope(slope, in_scale, out_scale);
-                gna_pwl[1].slope = FLOAT_TO_INT16(s.slope * s.slope_scale);
-                gna_pwl[1].xBase = gna_pwl[1].xBase | s.slope_scale_index;
-                print_segment((int32_t)(gna_pwl[1].xBase & XBASEMASK) / in_scale, gna_pwl[1].yBase / out_scale, 1.0);
-            } else {
-                double min_x_val = -pwl[0].b / pwl[0].m;
-                double max_x_val = (y_max/out_scale - pwl[pwl_size - 2].b) / pwl[pwl_size - 2].m;
-                double min_y_val = fun.fqParams.set ? pwl[0].beta : 0;
-                double max_y_val = fun.fqParams.set ? pwl.front().beta : y_max / out_scale;
-                gna_pwl = create_multisegment_gna_pwl(pwl, in_scale, out_scale, min_x_val, max_x_val, min_y_val, max_y_val,
-                    fun.fqParams.set, true);
-                break;
-            }
-            break;
-        }
         default:
             THROW_GNA_EXCEPTION << "Unexpected function activation!" << fun;
     }
     insert_extra_pwl_segments(gna_pwl, y_min, y_max);
 }
+
+template<typename T>
+static void make_gna_pwl(const T* m,
```

Comment:
Could you please add the detailed description for this function, the arguments names are not clear now